### PR TITLE
Android API Level 31 deprecation fixes

### PR DESCRIPTION
**Pull request by florissmit1**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Bug fix

### :arrow_heading_down: What is the current behavior?
Currently when changing the `compileSdkVersion` in the `build.gradle` file from `29` to `31`, the `flutter build appbundle` will produce some deprecation warnings. 

### :new: What is the new behavior (if this is a feature change)?
The deprecation warnings are solved by replacing the deprecated functionality for users using `compileSdkVersion 29`. Also upped the `compileSdkVersion` to `31`.

### :boom: Does this PR introduce a breaking change?
No

### :bug: Recommendations for testing
Does not apply

### :memo: Links to relevant issues/docs

Resolves #830.

Documentation:
https://developer.android.com/reference/android/location/LocationProvider
https://developer.android.com/reference/android/location/Location#isFromMockProvider()

### :thinking: Checklist before submitting

- [ ] I made sure all projects build.
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy](https://dart.dev/tools/pub/versioning).
- [ ] I updated CHANGELOG.md to add a description of the change.
- [ ] I followed the style guide lines ([code style guide](https://github.com/Baseflow/flutter-geolocator/blob/master/CONTRIBUTING.md)).
- [ ] I updated the relevant documentation.
- [ ] I rebased onto current `master`.

### CODE REVIEW

File: geolocator_android/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## 1.0.1
+- Fixed the deprecation warnings/errors which caused the `flutter build appbundle` to fail when using `compileSdkVersion 29`
```
Comment (mvanbeusekom) [inline geolocator_android/CHANGELOG.md:2]: Should have an empty line between the version number and the description as inline with the rest of the document. Also the sentence should always end with a period.

File: geolocator_android/android/src/main/java/com/baseflow/geolocator/location/LocationManagerClient.java [hunk 2]
```diff
@@ -132,14 +132,15 @@ public synchronized void onLocationChanged(Location location) {
...
-    if (status == LocationProvider.AVAILABLE) {
-      onProviderEnabled(provider);
-    } else if (status == LocationProvider.OUT_OF_SERVICE) {
-      onProviderDisabled(provider);
-    }
+      if (status == android.location.LocationProvider.AVAILABLE) {
+          onProviderEnabled(provider);
+      } else if (status == android.location.LocationProvider.OUT_OF_SERVICE) {
+          onProviderDisabled(provider);
+      }
```
Comment (mvanbeusekom) [inline geolocator_android/android/src/main/java/com/baseflow/geolocator/location/LocationManagerClient.java:143]: Formatting isn't consistent with the rest of the file (indents are with 2 spaces, not 4).

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]

**Comment by ukieTux** [comment]
How about this issue ?
Because it's will be conflict with permission-handler library if you using both library in one project

**Comment by mvanbeusekom** [comment]
> How about this issue ? Because it's will be conflict with permission-handler library if you using both library in one project

I will take some time and move this a long, should be a release later today.

**Comment by ukieTux** [comment]
> I will take some time and move this a long, should be a release later today.

Great, thanks..

**Review by mvanbeusekom — APPROVED** [review]
LGTM

**Comment by mvanbeusekom** [comment]
@ukieTux, I just release version 2.0.0 of the `geolocator_android` package and version 7.7.0 of the `geolocator` which now has support for Android SDK 31 (Android 12). 

Please not that to use this you will have to update your `android/app/build.gradle` file and set the `compileSdkVersion` to `31` (this is also documented in the CHANGELOG of both packages).

**Comment by ukieTux** [comment]
Great, thanks @mvanbeusekom 🙌

**Pull request merged**